rotkehlchen/chain/ethereum/modules/shutter/decoder.py

```diff
@@ -0,0 +1,127 @@
+            protocol=CPT_SHUTTER,
+            proposals_url='https://shutterdao.discourse.group/c/shutter-dao/dao-proposals/15',
+        )
+        self.shu = A_SHU.resolve_to_evm_token()
+
+    def _decode_shutter_claim(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] != SHU_CLAIMED:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = vesting_pool_address = None
+        for tx_log in context.all_logs:
+            if (
+                tx_log.address == self.shu.evm_address and
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                hex_or_bytes_to_address(tx_log.topics[1]) == SHUTTER_AIDROP_CONTRACT
+            ):
+                amount = token_normalized_value(token_amount=hex_or_bytes_to_int(tx_log.data), token=self.shu)  # noqa: E501
+                vesting_pool_address = hex_or_bytes_to_address(tx_log.topics[2])
+                break
+
+        if amount is None:
+            log.error(f'Could not find the amount of the SHU claimed in {context.transaction.tx_hash.hex()}')  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+        if vesting_pool_address is None:
+            log.error(f'Could not find the vesting pool address of the SHU claimed in {context.transaction.tx_hash.hex()}')  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        return DecodingOutput(event=self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=self.shu,
+            balance=Balance(amount=FVal(amount)),
+            location_label=hex_or_bytes_to_address(context.tx_log.topics[2]),
+            notes=f'Claim {amount} SHU vested from shutter airdrop and deposit it to {vesting_pool_address}',  # noqa: E501
+            counterparty=CPT_SHUTTER,
+            address=context.transaction.to_address,
+        ))
+
+    def _decode_delegation_change(self, context: DecoderContext) -> DecodingOutput:
+        """Decode a change in the delegated address"""
+        from_delegate = hex_or_bytes_to_address(context.tx_log.topics[2])
+        to_delegate = hex_or_bytes_to_address(context.tx_log.topics[3])
+
+        if self.base.is_tracked(from_delegate):
+            event_address = from_delegate
+        elif self.base.is_tracked(to_delegate):
+            event_address = to_delegate
+        else:
+            return DEFAULT_DECODING_OUTPUT
+
+        return DecodingOutput(
+            event=self.base.make_event_from_transaction(
+                transaction=context.transaction,
+                tx_log=context.tx_log,
+                event_type=HistoryEventType.INFORMATIONAL,
+                event_subtype=HistoryEventSubType.GOVERNANCE,
+                asset=self.shu,
+                balance=Balance(),
+                location_label=event_address,
```

Comment:
What are you doing man? I don't think you understand what location label is. 

This is the address that the event touches.It's the address I filter by when I filter my events by address.

So if I delegate to someone then this is my address. The way you made it is ... either the one I used to delegate to, or the one I will delegate to. Which is sort of stupid .... no?

User `user_address` here which should be `transaction.from_address`. Since it's used nowhere else no need to make an extra variable for it I guess.